plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxRecordSetProvider.java

```diff
@@ -169,9 +168,7 @@ public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorS
             }
             else {
                 List<Integer> selectedColumns = calculateSelectedColumns(tableHandle.getColumnHandles(), getColumnNames(columns));
-                rows = tableHandle.getObjectNames().stream()
-                        .flatMap(objectName -> jmxHistoricalData.getRows(objectName, selectedColumns).stream())
-                        .collect(toImmutableList());
+                rows = ImmutableList.copyOf(jmxHistoricalData.getRows(tableHandle.getTableName().getTableName(), selectedColumns));
```

Comment:
I think - we don't need this change. A JMXTableName could be a regex and it could match with a bunch of JMXObjects to we need to do a flatmapping and fetch the information for all the objects which matches the query pattern.